ArjunaJTA/jms/src/main/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelper.java

```diff
@@ -88,64 +88,55 @@ public boolean initialise(String properties) {
 
     @Override
     public void start(Xid xid, int i) throws XAException {
-        assert delegate != null : "Recovery scan has to be started";
-
+        assertIfRecoveryScanWasStarted();
         delegate.start(xid, i);
     }
 
     @Override
     public void end(Xid xid, int i) throws XAException {
-        assert delegate != null : "Recovery scan has to be started";
-
+        assertIfRecoveryScanWasStarted();
         delegate.end(xid, i);
     }
 
     @Override
     public int prepare(Xid xid) throws XAException {
-        assert delegate != null : "Recovery scan has to be started";
-
+        assertIfRecoveryScanWasStarted();
         return delegate.prepare(xid);
     }
 
     @Override
     public void commit(Xid xid, boolean b) throws XAException {
-        assert delegate != null : "Recovery scan has to be started";
-
+        assertIfRecoveryScanWasStarted();
```

Comment:
So maybe change the signature of assertIfRecoveryScanWasStarted(int xacode) so that you can tailor the error code based on the method where the error was detected.